ivy/functional/frontends/torch/nn/functional/loss_functions.py

```diff
@@ -303,6 +303,52 @@ def nll_loss(
     return ret
 
 
+@to_ivy_arrays_and_back
+def gaussian_nll_loss(
+    input,
+    target,
+    var,
+    full=False,
+    eps=1e-6,
+    reduction="mean"
+):
+
+    if var.shape != input.shape:
+        if input.shape[:-1] == var.shape:
+            var = torch_frontend.unsqueeze(var, dim=2)
+        elif input.shape[:-1] == var.shape[:-1] and var.shape[-1] == 1:
+            pass
+        else:
+            raise ivy.utils.exceptions.IvyError(
+                "var is of incorrect size"
+            )
+
+    if reduction is not None and reduction != "mean" and reduction != "sum":
+        raise ivy.utils.exceptions.IvyError(
+            f"{reduction} is not valid"
+        )
+
+    if ivy.any(var < 0):
+        raise ivy.utils.exceptions.IvyError(
+            "var has negative entry/entries"
+        )
+
+    var = torch_frontend.tensor(ivy.array(var, copy=True))
+    ivy.set_with_grads(False)
+    var = torch_frontend.clamp(var, min=eps)
+    ivy.unset_with_grads()
```

Comment:
Continuing where we left off in the previous PR, I don't think it will be necessary to do exactly what the source code does here with the gradients. You could simply do `var = torch_frontend.clamp(var, min=eps)`, but I would suggest `var = ivy.maximum(var, eps)` because it's more efficient to use an ivy function instead of a frontend one.

Reply:
Fixed these.